quipucords/scanner/satellite/connect.py

```diff
@@ -38,50 +38,45 @@ def __init__(self, scan_job, scan_task):
-            error_message = 'Satellite version is unknown. '
-            error_message += 'Connect scan failed for %s.' % self.scan_task
-            return error_message, ScanTask.FAILED
-
+        """Scan Satellite for system connection data."""
         try:
-            status_code, api_version = utils.status(self.scan_task,
-                                                    satellite_version)
+            status_code, api_version, satellite_version = \
+                utils.status(self.scan_task)
             if status_code == 200:
                 api = create(satellite_version, api_version,
                              self.scan_task)
                 if not api:
                     error_message = 'Satellite version %s with '\
                         'api version %s is not supported.\n' %\
                         (satellite_version, api_version)
-                    error_message += 'Connect scan failed for %s.' % \
-                        self.scan_task
+                    error_message += 'Connect scan failed for %s. ' % \
+                        self.source.name
                     return error_message, ScanTask.FAILED
                 api.host_count()
                 api.hosts()
             else:
-                error_message = 'Connect scan failed for %s.' % self.scan_task
+                error_message = 'Connect scan failed for source %s.' \
+                    % self.source.name
                 return error_message, ScanTask.FAILED
+        except SatelliteAuthException as auth_error:
+            error_message = 'Satellite error encountered: %s. ' % auth_error
+            error_message += 'Connect scan failed for source %s.' \
+                % self.source.name
+            return error_message, ScanTask.FAILED
         except SatelliteException as sat_error:
-            error_message = 'Satellite error encountered: %s\n' % sat_error
-            error_message += 'Connect scan failed for %s.' % self.scan_task
+            error_message = 'Satellite error encountered: %s. ' % sat_error
+            error_message += 'Connect scan failed for source %s.' \
+                % self.source.name
             return error_message, ScanTask.FAILED
         except exceptions.ConnectionError as conn_error:
-            error_message = 'Satellite error encountered: %s\n' % conn_error
-            error_message += 'Connect scan failed for %s.' % self.scan_task
+            error_message = 'Satellite error encountered: %s. ' % conn_error
+            error_message += 'Connect scan failed for source %s.' \
+                % self.source.name
             return error_message, ScanTask.FAILED
         except TimeoutError as timeout_error:
-            error_message = 'Satellite error encountered: %s\n' % timeout_error
-            error_message += 'Connect scan failed for %s.' % self.scan_task
-            return error_message, ScanTask.FAILED
-        except Exception as unknown_error:  # pylint: disable=broad-except
```

Comment:
I think the general exception could be useful now that we don't know for sure the version of Satellite we are talking too.  The job handle the exception this way:
```
message = 'FATAL ERROR. %s' % str(error)
self.scan_job.fail(message)
raise error
```

Would it be useful to log critical information and then raise again?

Reply:
I think that is a good point, but maybe we should open an issue that captures the job information in the Job's general exception capture.